spikeinterface/core/basesorting.py

```diff
@@ -51,10 +51,51 @@ def get_unit_spike_train(self,
                              segment_index: Union[int, None] = None,
                              start_frame: Union[int, None] = None,
                              end_frame: Union[int, None] = None,
+                             return_times: bool = False
                              ):
         segment_index = self._check_segment_index(segment_index)
-        S = self._sorting_segments[segment_index]
-        return S.get_unit_spike_train(unit_id=unit_id, start_frame=start_frame, end_frame=end_frame).astype("int64")
+        segment = self._sorting_segments[segment_index]
+        spike_train = segment.get_unit_spike_train(
+            unit_id=unit_id, start_frame=start_frame, end_frame=end_frame).astype("int64")
+        if return_times:
+            if self.has_time_vector:
+                times = self.get_times(segment_index=segment_index)
+                return times[spike_train]
+            else:
+                return spike_train / self.get_sampling_frequency()
```

Comment:
You should also handle t_start here if t_start is not None.

Reply:
changed it so it gets here only if there is no registered recording